rcl_action/test/rcl_action/test_types.cpp

```diff
@@ -0,0 +1,151 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <gtest/gtest.h>
+
+#include "rcl_action/types.h"
+
+
+TEST(TestActionTypes, test_get_zero_inititalized_goal_info)
+{
+  rcl_action_goal_info_t goal_info = rcl_action_get_zero_initialized_goal_info();
+  ASSERT_EQ(sizeof(goal_info.uuid) / sizeof(uint8_t), 16u);
+  for (int i = 0; i < 16; ++i) {
+    EXPECT_EQ(goal_info.uuid[i], 0u);
+  }
+  EXPECT_EQ(goal_info.stamp.sec, 0);
+  EXPECT_EQ(goal_info.stamp.nanosec, 0u);
+
+  // Modify the first, get another zero initialized goal info struct
+  for (int i = 0; i < 16; ++i) {
+    goal_info.uuid[i] = static_cast<uint8_t>(i);
+  }
+  goal_info.stamp.sec = 1234;
+  goal_info.stamp.nanosec = 4567u;
+  rcl_action_goal_info_t another_goal_info = rcl_action_get_zero_initialized_goal_info();
+  for (int i = 0; i < 16; ++i) {
+    EXPECT_EQ(goal_info.uuid[i], i);
+    EXPECT_EQ(another_goal_info.uuid[i], 0u);
+  }
+  EXPECT_EQ(goal_info.stamp.sec, 1234);
+  EXPECT_EQ(goal_info.stamp.nanosec, 4567u);
+  EXPECT_EQ(another_goal_info.stamp.sec, 0);
+  EXPECT_EQ(another_goal_info.stamp.nanosec, 0u);
+}
+
+TEST(TestActionTypes, test_get_zero_initialized_goal_status_array)
+{
+  rcl_action_goal_status_array_t status_array =
+    rcl_action_get_zero_initialized_goal_status_array();
+  ASSERT_EQ(sizeof(status_array.status_list) / sizeof(rcl_action_goal_status_t), 0u);
```

Comment:
@jacobperron what's this intent of this test? Won't `sizeof(status_array.status_list)` be always the same?